### PR TITLE
PWN-4398 - Actions BottomSheet

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4398

## Description of Work

Made an actions bottomsheet to interact with main button

## Screenshots (Optional)

Android Emulator - Pixel_5_API_30:5554 2022-08-04 18-56-28

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/ActionsBottomSheet.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        ) = MainActionsBottomSheet().withArgs(
+            EXTRA_REQUEST_KEY to requestKey,
+            EXTRA_RESULT_KEY to resultKey
+        ).show(fm, MainActionsBottomSheet::javaClass.name)
+    }
+
+    private val resultKey: String by args(EXTRA_RESULT_KEY)
+    private val requestKey: String by args(EXTRA_REQUEST_KEY)
+
+    private val binding: DialogMainActionsBinding by viewBinding()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/ActionsBottomSheet.kt:37]: layout inflates two times here I think
in by viewBinding()
and in inflater.inflate

File: app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/ActionsBottomSheet.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    private fun ViewActionItemBinding.setResultClickListener(action: MainAction) {
+        viewActionRoot.setOnClickListener {
+            setFragmentResult(requestKey, bundleOf(resultKey to action))
+            dismissAllowingStateLoss()
+        }
+    }
+}
+
+enum class MainAction {
+    Buy, Receive, Trade, Send
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/ActionsBottomSheet.kt:83]: UPPER_CASE?

File: app/src/main/res/layout/dialog_main_actions.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        android:layout_width="32dp"
+        android:layout_height="4dp"
+        android:layout_marginTop="6dp"
+        android:background="@drawable/shape_bottomsheet_pointer"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <TextView
+        android:id="@+id/viewBottomSheetTitle"
```
Comment (gslevinkov) [inline app/src/main/res/layout/dialog_main_actions.xml:22]: textViewBottomSheetTitle

File: app/src/main/res/layout/dialog_main_actions.xml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        style="@style/UiKit.TextAppearance.SemiBold.Title3"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginHorizontal="20dp"
+        android:layout_marginTop="16dp"
+        android:gravity="center"
+        android:text="@string/main_actions_title"
+        android:textColor="@color/night"
```
Comment (gslevinkov) [inline app/src/main/res/layout/dialog_main_actions.xml:30]:
```suggestion
        android:textColor="@color/text_night"
```

File: app/src/main/res/layout/dialog_main_actions.xml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        android:layout_marginStart="18dp"
+        android:layout_marginTop="20dp"
+        android:layout_marginEnd="8dp"
+        app:layout_constraintBottom_toBottomOf="@+id/viewBottomSheetActionReceive"
+        app:layout_constraintEnd_toStartOf="@+id/viewBottomSheetActionReceive"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/viewBottomSheetTitle" />
+
+    <include
+        android:id="@+id/viewBottomSheetActionReceive"
```
Comment (gslevinkov) [inline app/src/main/res/layout/dialog_main_actions.xml:49]: I think we can remove bottomSheet from ids

File: app/src/main/res/layout/view_action_item.xml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    android:id="@+id/viewActionRoot"
+    android:background="@drawable/bg_main_action_rounded"
+    android:foreground="@drawable/ripple_button_large"
+    android:gravity="center_vertical"
+    android:orientation="vertical"
+    android:paddingHorizontal="20dp"
+    android:paddingVertical="16dp">
+
+    <ImageView
+        android:id="@+id/imageViewActionIcon"
```
Comment (gslevinkov) [inline app/src/main/res/layout/view_action_item.xml:17]: Icon is redundant

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/ActionsBottomSheet.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/ActionsBottomSheet.kt:37]: yes but do you know what we need to do here?

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/ActionsBottomSheet.kt:37]: override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View = binding.root ?

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/ActionsBottomSheet.kt:37]: binding.root  - crash =)

**Review by kuznetsov — APPROVED** [review]

**Pull request merged**